### PR TITLE
Introduce rawDataBucket

### PR DESCRIPTION
**Pull request by davidmirror-ops**
## Tracking issue
#4291 

## Why are the changes needed?
Enables users to specify a separate (from metadata) Raw Data bucket in the form of a userDataBucket parameter in flyte-core. 

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

### CODE REVIEW

File: charts/flyte-core/values-eks.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ userSettings:
   dbPassword: <DB_PASSWORD>
   rdsHost: <RDS_HOST>
   bucketName: <BUCKET_NAME>
+  userBucketName: <RAW_DATA_BUCKET_NAME>
```
Comment (pingsutw) [inline charts/flyte-core/values-eks.yaml:8]: would it be better to call it `rawDataBucketName`?

File: deployment/eks/flyte_aws_scheduler_helm_generated.yaml [hunk 2]
```diff
@@ -179,6 +179,8 @@ data:
       signedUrls:
         durationMinutes: 3
   storage.yaml: | 
+    propeller:
+      rawoutput-prefix: s3://<RAW_DATA_BUCKET_NAME>/data
```
Comment (pingsutw) [inline deployment/eks/flyte_aws_scheduler_helm_generated.yaml:183]: why do we need `/data` here?

File: deployment/sandbox/flyte_helm_generated.yaml [hunk 3]
```diff
@@ -498,6 +500,8 @@ data:
       profiler-port: 10254
       storage-prefix: metadata/datacatalog
   storage.yaml: | 
+    propeller:
+      rawoutput-prefix:
```
Comment (pingsutw) [inline deployment/sandbox/flyte_helm_generated.yaml:504]: IIRC, `rawoutput-prefix` can't be empty. does it work in the sandbox?

File: deployment/eks/flyte_aws_scheduler_helm_generated.yaml (hunk 2 above)
Comment (davidmirror-ops) [reply deployment/eks/flyte_aws_scheduler_helm_generated.yaml:183]: Not needed, it's the way binary handles it currently. I thought it'd be nice to align them

**Pull request closed without merging**